src/bank.rs

```diff
@@ -200,18 +207,23 @@ impl Bank {
         }
     }
 
-    fn reserve_signature_with_last_id(&self, signature: &Signature, last_id: &Hash) -> Result<()> {
-        if let Some(entry) = self
-            .last_ids_sigs
-            .write()
-            .expect("'last_ids' read lock in reserve_signature_with_last_id")
-            .get_mut(last_id)
-        {
-            return Self::reserve_signature(&mut entry.0, signature);
+    fn reserve_signature_with_last_id(
+        last_ids_sigs: &mut HashMap<Hash, (SignatureStatusMap, u64)>,
```

Comment:
Please add a ticket to move this HashMap out of the Bank. This should clearly be a method, not a function.

Reply:
#1412